trio/_core/_run.py

```diff
@@ -1173,7 +1164,10 @@ def start_soon(
         GLOBAL_RUN_CONTEXT.runner.spawn_impl(async_fn, args, self, name)
 
     async def start(
-        self, async_fn: _NurseryStartFunc[StatusT], *args: object, name: object = None
+        self,
+        async_fn: Callable[..., Awaitable[object]],
+        *args: object,
+        name: object = None,
```

Comment:
So this drops the check for it accepting a `task_status` parameter. If that's planned to be added back with `ParamSpec`/`TypeVarTuple` it probably warrants a `#TODO` explicitly mentioning that. Or is there a reason for dropping it?

Reply:
Yeah it needs TypeVarTuple. That protocol would only match callables that actually themselves use `*args`, which is not ideal.